pkg/cli/manifest/registermanifest.go

```diff
@@ -378,3 +358,60 @@ func ValidateManifest(ctx context.Context, path string) (resourceProvider *Resou
 
 	return resourceProvider, nil
 }
+
+// extractLocationInfo extracts location name and address from resource provider
+func extractLocationInfo(resourceProvider ResourceProvider) (string, string) {
+	var locationName string
+	var address string
+
+	if resourceProvider.Location == nil {
+		locationName = v1.LocationGlobal
+	} else {
+		for name, addr := range resourceProvider.Location {
+			locationName = name
+			address = addr
+			break // We support one location per resourceProvider
+		}
+	}
+	return locationName, address
+}
+
+// createResourceProviderResource creates the resource provider entity
+func createResourceProviderResource(ctx context.Context, clientFactory *v20231001preview.ClientFactory, planeName string, resourceProvider ResourceProvider, locationName string, logger func(format string, args ...any)) error {
```

Comment:
I see `extractLocationInfo` has test scenarios. Do we need unit tests for `createResourceProviderResource` and `createLocationResource`

Reply:
Excellent suggestion. Those functions are [covered](https://app.codecov.io/gh/radius-project/radius/pull/10317/blob/pkg/cli/manifest/registermanifest.go?dropdown=coverage#L380) by other tests, but I added some addtional tests that directly target them.